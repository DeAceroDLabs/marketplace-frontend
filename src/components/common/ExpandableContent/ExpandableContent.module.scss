@use 'styles/colors';
@use 'styles/typography';

.container{
    margin: 37px 0px;
}

.title-arrow-container {
    font-weight: typography.$bold;
    font-size: typography.$header-s;
    display: flex;
}