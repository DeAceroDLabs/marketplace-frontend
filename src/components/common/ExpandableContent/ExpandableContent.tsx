import styles from "./ExpandableContent.module.scss";
import { useState } from "react";
import ArrowDropDownIcon from "@mui/icons-material/ArrowDropDown";
import ArrowDropUpIcon from "@mui/icons-material/ArrowDropUp";
import Button from "components/common/Button";

interface ExpandableContentProps {
  title: string;
  children?: React.ReactNode;
  isOpen?: boolean;
}

const ExpandableContent: React.FC<ExpandableContentProps> = ({
  title,
  children,
  isOpen = false,
}) => {
  const [visible, setVisible] = useState(isOpen);
  const handleVisibility = () => {
    setVisible(!visible);
  };
  return (
    <div className={styles.container}>
      <div className={styles["title-arrow-container"]}>
        <div>{title}</div>
        <Button color="transparent-black" action={() => handleVisibility()}>
          {visible && <ArrowDropUpIcon />}
          {!visible && <ArrowDropDownIcon />}
        </Button>
      </div>
      <hr className={styles.line} />
      {visible && children}
    </div>
  );
};

export default ExpandableContent;
