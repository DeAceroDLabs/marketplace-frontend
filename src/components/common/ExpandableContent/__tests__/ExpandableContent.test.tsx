import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import ExpandableContent from "../ExpandableContent";

describe("ExpandableContent", () => {
  const view = render(
    <ExpandableContent title="Mock categoryName">
      <div>Mock content</div>
    </ExpandableContent>
  );

  it("renders ExpandableContent with no issue", () => {
    expect(view).toMatchSnapshot();
  });

  it("arrow button calls handleVisibility", async () => {
    render(
      <ExpandableContent title="Mock categoryName">
      <div>Mock content</div>
    </ExpandableContent>
    );
    const arrowButton = screen.getByRole("button");
    fireEvent.click(arrowButton);
    await waitFor(() => {
      expect(screen.getByText("Mock content")).toBeInTheDocument();
    });
  });
});
