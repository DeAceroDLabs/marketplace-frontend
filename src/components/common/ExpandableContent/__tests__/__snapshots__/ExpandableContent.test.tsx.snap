// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandableContent renders ExpandableContent with no issue 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="title-arrow-container"
        >
          <div>
            Mock categoryName
          </div>
          <button
            class="button transparent-black activeClass"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m7 10 5 5 5-5z"
              />
            </svg>
          </button>
        </div>
        <hr
          class="black"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="title-arrow-container"
      >
        <div>
          Mock categoryName
        </div>
        <button
          class="button transparent-black activeClass"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m7 10 5 5 5-5z"
            />
          </svg>
        </button>
      </div>
      <hr
        class="black"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
